### DISCUSSION TITLE
Question about middleware

### DISCUSSION THREAD

**Discussion by devleejb** [vercel/community#1192] [Help] [answered]
### Summary

Hi,

I'm implementing middleware to block access without authentication.
I wrote middleware.ts at project top level and it works.
However, when the request body size is large, the request doesn't work properly. (The request doesn't reach to pages or api.)
Is there any option for size limit for middleware like in api? (Like below code)

```
export const config = {
    api: {
        bodyParser: {
            sizeLimit: "10mb", // Set desired value here
        },
    },
};
```

Or, is there any other method to implement middleware globally?

Thank you.

**Comment by swarnava** [comment, marked as answer]
Hi,

your request body shouldn't exceed than 4MB. You can check our limit here: https://vercel.com/docs/concepts/functions/edge-middleware/limitations#limits-on-requests
